Firmware/Marlin_main.cpp

```diff
@@ -7238,6 +7238,87 @@ SERIAL_PROTOCOLPGM("\n\n");
 
     }break;
     #endif // FWRETRACT
+    /*!
+    ### M214 - Set Arc configuration values (Use M500 to store in eeprom)
+
+    #### Usage
+
+        M214 [N] [S] [R] [F]
+
+    #### Parameters
+    - `N` - A float representing the max and default millimeters per arc segment.  Must be greater than 0.
+    - `S` - A float representing the minimum allowable millimeters per arc segment.  Set to 0 to disable
+    - `R` - An int representing the minimum number of segments per arcs of any radius,
+            except when the results in segment lengths greater than or less than the minimum
+            and maximum segment length.  Set to 0 to disable.
+    - 'F' - An int representing the number of segments per second, unless this results in segment lengths
+            greater than or less than the minimum and maximum segment length.  Set to 0 to disable.
+    */
+    case 214: //!@n M214 - Set Arc Parameters (Use M500 to store in eeprom) N<MM_PER_ARC_SEGMENT> S<MIN_MM_PER_ARC_SEGMENT> R<MIN_ARC_SEGMENTS> F<ARC_SEGMENTS_PER_SEC>
+    {
+        // Extract N
+        float n = cs.mm_per_arc_segment;
+        float s = cs.min_mm_per_arc_segment;
+        int r = cs.min_arc_segments;
+        int f = cs.arc_segments_per_sec;
+        // Extract N
+        if (code_seen('N'))
+        {
+            n = code_value();
+            if (n <= 0 || (s != 0 && n <= s))
+            {
```

Comment:
No need to write a message so complex to the user. It's enough if you just break from the command processing switch early. At most tell the user that a wrong argument was provided.

Reply:
Perhaps something like this:

messages.c - Add invalid argument value message
```
const char MSG_INVALID_ARGUMENT_VALUE[] PROGMEM_N1 = "Invalid argument value: "; ////
```

Marlin_main.cpp
```
                SERIAL_ECHO_START;
                SERIAL_ECHORPGM(MSG_INVALID_ARGUMENT_VALUE);
                SERIAL_ECHO("S");
                SERIAL_ECHOLNPGM("");
```

I couldn't find an existing invalid argument value message.  I could easily just break as you mentioned if this is not acceptable.
??

Reply:
For one it's repeating code. Also, no other command does this and I don't think it's really necessary. Breaking directly is probably good enough in this case.